controllers/whoToFollow/whoToFollowV2.js

```diff
@@ -220,22 +220,23 @@ module.exports = async (req, res) => {
                             a.is_anonymous = false
                             AND a.is_banned = false
                             AND a.blocked_by_admin = false
-                            AND a.karma_score > 30
+                            AND a.karma_score > 45
                             AND b.topic_id NOT IN (:topics)
                             AND a.user_id NOT IN (:allUserIds)
                           GROUP BY a.user_id
                           ORDER BY 
-                            a.last_active_at DESC,
-                            CASE WHEN a.profile_pic_path != '%default-profile-picture%' THEN 0 ELSE 1 END,
+                            CASE WHEN a.last_active_at > now() - interval '7' day THEN 0 ELSE 1 END,
+                            CASE WHEN a.profile_pic_path NOT LIKE '%default-profile-picture%' THEN 0 ELSE 1 END,
                             a.followers_count DESC
-                          LIMIT :limit`;
+                          LIMIT :limit OFFSET :offset`;
 
       const userOtherTopicFollowerQueryResult = await sequelize.query(otherTopicsQuery, {
         type: sequelize.QueryTypes.SELECT,
         replacements: {
           topics,
           allUserIds,
-          limit
+          limit,
+          offset
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, the increase in the karma score threshold and the addition of the condition related to the last active time should refine the user recommendations based on topics. The introduction of the offset in the pagination could improve the efficiency and accuracy of data retrieval. However, ensure that the offset is being calculated correctly elsewhere in your code to avoid skipping or repeating users across pages.

```diff
- AND a.karma_score > 30
+ AND a.karma_score > 45
- a.last_active_at DESC,
+ CASE WHEN a.last_active_at > now() - interval '7' day THEN 0 ELSE 1 END,
- LIMIT :limit`;
+ LIMIT :limit OFFSET :offset`;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->